src/ansible_grid.c

```diff
@@ -5191,17 +5280,22 @@ void handler_ESGridKey(s32 data) {
             es_runes = z;
         } else if (y == 7) { // voices
             es_voices = z;
+        } else if (y > 7 && y < 16) { // 256-mode pattern change hotkeys
+            // jump between patterns 0-7 or 8-15, whichever half is currently active
+            e.p_select = (e.p_select & 0x8) + (15 - y);
         }
 
-        if (!es_edge) {
-            monomeFrameDirty++;
-            return;
-        }
+        monomeFrameDirty++;
+        return;
     }
 
-    if (es_runes) {
+    if (es_runes || (monome_size_y() == 16 && y > 8 && !es_edge && !es_voices && (es_view == es_patterns || es_view == es_patterns_held))) {
```

Comment:
don't need `monome_size_y() == 16` here - it's covered by `y > 8`

Reply:
Yep, I thought it made the intent clearer, but you're right, it's not worth the extra check.